py/pysparkling/context.py

```diff
@@ -119,12 +119,20 @@ def getOrCreate(spark_context, conf = None):
         # Create H2O REST API client
         h2o.connect(ip=h2o_context._client_ip, port=h2o_context._client_port)
         h2o_context.is_initialized = True
+        # Stop h2o when running standalone pysparkling scripts and the user does not explicitly close h2o
+        atexit.register(lambda: h2o_context.stop())
         return h2o_context
 
     def stop(self):
-        warnings.warn("H2OContext stopping is not yet fully supported...")
+        h2o.cluster().shutdown()
         self._jhc.stop(False)
 
+    def __del__(self):
+        self.stop()
+
+    def __exit__(self):
+        self.stop()
```

Comment:
Maybe, you should separate `atexit` functionality from regular `H2OContext.stop`. 

Something like:

```python
import atexit

@atexit.register
def goodbye():
    h2o.cluster().shutdown()
    self._jhc.stop(False)
```